src/metabase/api/common.clj

```diff
@@ -261,16 +262,50 @@
                                   (ns-name *ns*) fn-name)))
       (assoc parsed :fn-name fn-name, :route route, :docstr docstr))))
 
+(defn validate-param-values
+  "Log a warning if the request body contains any parameters not included in `expected-params` (which is presumably
+  populated by the defendpoint schema)"
+  [{route :compojure/route body :body} expected-params]
+  (when (map? body)
+    (let [extraneous-params (set/difference (set (keys body))
+                                            (set expected-params))]
+      (when (seq extraneous-params)
+        (log/warn (format (str "Unexpected parameters at %s: %s\nPlease add them to the schema or remove them from the"
```

Comment:
You could use `log/warnf` which combines `log`/warn` + `format`

It's always a good idea to add a little note saying why you're not i18n'ing something. If this is mean to be dev-facing only then that's fine, just add a note to that effect so people don't think you simply forgot to use `trs`. On that note: **should** this only be enabled in `:dev` mode? Suppose the FE client is sending a key and we accept it but don't have a schema to validate it: should we be filling our user's logs with an untranslated warning they can't do anything about?

Reply:
I guess maybe you could do it in `:dev` + `:test` mode but not `:prod` so your test still works